--- cloud-init.py.orig	2012-07-13 16:32:36.000000000 +0200
+++ cloud-init.py	2012-07-13 16:32:50.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # vi: ts=4 expandtab
 #
 #    Copyright (C) 2009-2010 Canonical Ltd.
--- cloud-init-cfg.py.orig	2012-07-13 16:34:14.000000000 +0200
+++ cloud-init-cfg.py	2012-07-13 16:34:20.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # vi: ts=4 expandtab
 #
 #    Copyright (C) 2009-2010 Canonical Ltd.
--- cloud-init-query.py.orig	2012-07-13 16:35:18.000000000 +0200
+++ cloud-init-query.py	2012-07-13 16:35:59.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # vi: ts=4 expandtab
 #
 #    Copyright (C) 2009-2010 Canonical Ltd.
--- setup.py.orig	2012-07-13 16:35:31.000000000 +0200
+++ setup.py	2012-07-13 16:36:19.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # vi: ts=4 expandtab
 #
 #    Distutils magic for ec2-init
--- cloudinit/SshUtil.py.orig	2012-07-13 16:35:43.000000000 +0200
+++ cloudinit/SshUtil.py	2012-07-13 16:36:28.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # vi: ts=4 expandtab
 #
 #    Copyright (C) 2012 Canonical Ltd.
--- cloudinit/netinfo.py.orig	2012-07-13 16:35:50.000000000 +0200
+++ cloudinit/netinfo.py	2012-07-13 16:36:39.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # vi: ts=4 expandtab
 #
 #    Copyright (C) 2012 Canonical Ltd.
