Index: cloud-init-0.6.3-bzr532/systemd/cloud-config.service
===================================================================
--- cloud-init-0.6.3-bzr532.orig/systemd/cloud-config.service
+++ cloud-init-0.6.3-bzr532/systemd/cloud-config.service
@@ -10,5 +10,8 @@ ExecStart=/usr/bin/cloud-init-cfg all co
 RemainAfterExit=yes
 TimeoutSec=0
 
+# Output needs to appear in instance console output
+StandardOutput=tty
+
 [Install]
 WantedBy=multi-user.target
Index: cloud-init-0.6.3-bzr532/systemd/cloud-final.service
===================================================================
--- cloud-init-0.6.3-bzr532.orig/systemd/cloud-final.service
+++ cloud-init-0.6.3-bzr532/systemd/cloud-final.service
@@ -10,5 +10,8 @@ ExecStart=/usr/bin/cloud-init-cfg all fi
 RemainAfterExit=yes
 TimeoutSec=0
 
+# Output needs to appear in instance console output
+StandardOutput=tty
+
 [Install]
 WantedBy=multi-user.target
Index: cloud-init-0.6.3-bzr532/systemd/cloud-init-local.service
===================================================================
--- cloud-init-0.6.3-bzr532.orig/systemd/cloud-init-local.service
+++ cloud-init-0.6.3-bzr532/systemd/cloud-init-local.service
@@ -9,5 +9,8 @@ ExecStart=/usr/bin/cloud-init start-loca
 RemainAfterExit=yes
 TimeoutSec=0
 
+# Output needs to appear in instance console output
+StandardOutput=tty
+
 [Install]
 WantedBy=multi-user.target
Index: cloud-init-0.6.3-bzr532/systemd/cloud-init.service
===================================================================
--- cloud-init-0.6.3-bzr532.orig/systemd/cloud-init.service
+++ cloud-init-0.6.3-bzr532/systemd/cloud-init.service
@@ -10,5 +10,8 @@ ExecStart=/usr/bin/cloud-init start
 RemainAfterExit=yes
 TimeoutSec=0
 
+# Output needs to appear in instance console output
+StandardOutput=tty
+
 [Install]
 WantedBy=multi-user.target
