Index: cloud-init-0.7.5/cloudinit/sources/DataSourceAltCloud.py
===================================================================
--- cloud-init-0.7.5.orig/cloudinit/sources/DataSourceAltCloud.py
+++ cloud-init-0.7.5/cloudinit/sources/DataSourceAltCloud.py
@@ -42,7 +42,7 @@ CLOUD_INFO_FILE = '/etc/sysconfig/cloud-
 # Shell command lists
 CMD_DMI_SYSTEM = ['/usr/sbin/dmidecode', '--string', 'system-product-name']
 CMD_PROBE_FLOPPY = ['/sbin/modprobe', 'floppy']
-CMD_UDEVADM_SETTLE = ['/sbin/udevadm', 'settle', '--quiet', '--timeout=5']
+CMD_UDEVADM_SETTLE = ['/sbin/udevadm', 'settle', '--timeout=5']
 
 META_DATA_NOT_SUPPORTED = {
     'block-device-mapping': {},
