--- cloud-init-0.6.3-bzr532.orig/cloudinit/SshUtil.py	2012-03-05 14:05:34.000000000 -0500
+++ cloud-init-0.6.3-bzr532/cloudinit/SshUtil.py	2012-03-31 15:21:30.365271359 -0400
@@ -155,6 +155,8 @@
         akeys = ssh_cfg.get("AuthorizedKeysFile", "%h/.ssh/authorized_keys")
         akeys = akeys.replace("%h", pwent.pw_dir)
         akeys = akeys.replace("%u", user)
+        if not akeys.startswith('/'):
+            akeys = os.path.join(pwent.pw_dir, akeys)
         authorized_keys = akeys
     except Exception:
         authorized_keys = '%s/.ssh/authorized_keys' % pwent.pw_dir
