diff -Naur cloud-init-0.6.3-bzr532.orig/cloudinit/CloudConfig/cc_resizefs.py cloud-init-0.6.3-bzr532/cloudinit/CloudConfig/cc_resizefs.py
--- cloud-init-0.6.3-bzr532.orig/cloudinit/CloudConfig/cc_resizefs.py	2012-03-05 19:05:34.000000000 +0000
+++ cloud-init-0.6.3-bzr532/cloudinit/CloudConfig/cc_resizefs.py	2012-09-11 01:11:17.405125250 +0000
@@ -69,7 +69,7 @@
         (str(fstype).rstrip("\n"), os.major(st_dev), os.minor(st_dev)))
 
     if str(fstype).startswith("ext"):
-        resize_cmd = ['resize2fs', devpth]
+        resize_cmd = ['resize4fs', devpth]
     elif fstype == "xfs":
         resize_cmd = ['xfs_growfs', devpth]
     else:
