Index: cloud-init/setup.py
===================================================================
--- cloud-init.orig/setup.py
+++ cloud-init/setup.py
@@ -40,9 +40,8 @@ setup(name='cloud-init',
       data_files=[('/etc/cloud', glob('config/*.cfg')),
                   ('/etc/cloud/cloud.cfg.d', glob('config/cloud.cfg.d/*')),
                   ('/etc/cloud/templates', glob('templates/*')),
-                  ('/etc/init', glob('upstart/*.conf')),
                   ('/usr/share/cloud-init', []),
-                  ('/usr/lib/cloud-init', 
+                  ('/usr/libexec/cloud-init', 
                     ['tools/uncloud-init', 'tools/write-ssh-key-fingerprints']),
                   ('/usr/share/doc/cloud-init', filter(is_f,glob('doc/*'))),
                   ('/usr/share/doc/cloud-init/examples', filter(is_f,glob('doc/examples/*'))),
Index: cloud-init/cloudinit/CloudConfig/cc_keys_to_console.py
===================================================================
--- cloud-init.orig/cloudinit/CloudConfig/cc_keys_to_console.py
+++ cloud-init/cloudinit/CloudConfig/cc_keys_to_console.py
@@ -21,7 +21,7 @@ import subprocess
 frequency = per_instance
 
 def handle(name,cfg,cloud,log,args):
-    write_ssh_prog='/usr/lib/cloud-init/write-ssh-key-fingerprints'
+    write_ssh_prog='/usr/libexec/cloud-init/write-ssh-key-fingerprints'
     try:
         confp = open('/dev/console',"wb")
         subprocess.call(write_ssh_prog,stdout=confp)
Index: cloud-init/cloudinit/CloudConfig/cc_ssh.py
===================================================================
--- cloud-init.orig/cloudinit/CloudConfig/cc_ssh.py
+++ cloud-init/cloudinit/CloudConfig/cc_ssh.py
@@ -85,7 +85,7 @@ def handle(name,cfg,cloud,log,args):
     send_ssh_keys_to_console()
 
 def send_ssh_keys_to_console():
-    subprocess.call(('/usr/lib/cloud-init/write-ssh-key-fingerprints',))
+    subprocess.call(('/usr/libexec/cloud-init/write-ssh-key-fingerprints',))
 
 def apply_credentials(keys, user, disable_root, disable_root_opts=DISABLE_ROOT_OPTS, log=global_log):
     keys = set(keys)
