Index: cloud-init/cloudinit/CloudConfig/cc_timezone.py
===================================================================
--- cloud-init.orig/cloudinit/CloudConfig/cc_timezone.py
+++ cloud-init/cloudinit/CloudConfig/cc_timezone.py
@@ -38,13 +38,21 @@ def handle(name,cfg,cloud,log,args):
         log.debug("Invalid timezone %s" % tz_file)
         raise Exception("Invalid timezone %s" % tz_file)
 
-    try:
-        fp=open("/etc/timezone","wb")
-        fp.write("%s\n" % timezone)
-        fp.close()
-    except:
-        log.debug("failed to write to /etc/timezone")
-        raise
+    if os.path.exists("/etc/timezone"):
+        try:
+            fp=open("/etc/timezone","wb")
+            fp.write("%s\n" % timezone)
+            fp.close()
+        except:
+            log.debug("failed to write to /etc/timezone")
+            raise
+    elif os.path.exists("/etc/sysconfig/clock"):
+        try:
+            with open("/etc/sysconfig/clock", "w") as fp:
+                fp.write('ZONE="%s"\n' % timezone)
+        except:
+            log.debug("failed to write to /etc/sysconfig/clock")
+            raise
 
     try:
         shutil.copy(tz_file, "/etc/localtime")
