diff -uNr cloud-init.orig/cloudinit/config/cc_keys_to_console.py cloud-init/cloudinit/config/cc_keys_to_console.py
--- cloud-init.orig/cloudinit/config/cc_keys_to_console.py	2013-04-12 14:03:11.049761648 +0100
+++ cloud-init/cloudinit/config/cc_keys_to_console.py	2013-04-12 14:13:09.660280218 +0100
@@ -26,7 +26,7 @@
 frequency = PER_INSTANCE
 
 # This is a tool that cloud init provides
-HELPER_TOOL = '/usr/lib/cloud-init/write-ssh-key-fingerprints'
+HELPER_TOOL = '/usr/libexec/cloud-init/write-ssh-key-fingerprints'
 
 
 def handle(name, cfg, _cloud, log, _args):
diff -uNr cloud-init.orig/setup.py cloud-init/setup.py
--- cloud-init.orig/setup.py	2013-04-12 14:03:11.052761671 +0100
+++ cloud-init/setup.py	2013-04-12 14:08:12.828042068 +0100
@@ -45,7 +45,7 @@
 INITSYS_ROOTS = {
     'sysvinit': '/etc/rc.d/init.d',
     'sysvinit_deb': '/etc/init.d',
-    'systemd': '/etc/systemd/system/',
+    'systemd': '/usr/lib/systemd/system/',
     'upstart': '/etc/init/',
 }
 INITSYS_TYPES = sorted(list(INITSYS_ROOTS.keys()))
@@ -123,7 +123,7 @@
                   ('/etc/cloud/cloud.cfg.d', glob('config/cloud.cfg.d/*')),
                   ('/etc/cloud/templates', glob('templates/*')),
                   ('/usr/share/cloud-init', []),
-                  ('/usr/lib/cloud-init',
+                  ('/usr/libexec/cloud-init',
                     ['tools/uncloud-init',
                      'tools/write-ssh-key-fingerprints']),
                   ('/usr/share/doc/cloud-init',
