--- cloud-init-0.6.3-bzr532/cloudinit/DataSourceEc2.py.fqdn	2012-03-05 11:05:34.000000000 -0800
+++ cloud-init-0.6.3-bzr532/cloudinit/DataSourceEc2.py	2012-09-13 13:40:29.757623464 -0700
@@ -110,7 +110,7 @@ class DataSourceEc2(DataSource.DataSourc
             util.logexc(log)
             log.warn("Failed to get timeout, using %s" % timeout)
 
-        def_mdurls = ["http://169.254.169.254", "http://instance-data:8773"]
+        def_mdurls = ["http://169.254.169.254", "http://instance-data.:8773"]
         mdurls = mcfg.get("metadata_urls", def_mdurls)
 
         # Remove addresses from the list that wont resolve.
