Index: cloud-init/cloudinit/distros/rhel.py
===================================================================
--- cloud-init.orig/cloudinit/distros/rhel.py
+++ cloud-init/cloudinit/distros/rhel.py
@@ -208,7 +208,7 @@ class Distro(distros.Distro):
 
     def update_package_sources(self):
         self._runner.run("update-sources", self.package_command,
-                         ["update"], freq=PER_INSTANCE)
+                         ["clean", "expire-cache"], freq=PER_INSTANCE)
 
 
 # This class helps adjust the configobj
