Index: cloud-init-0.6.3-bzr532/systemd/cloud-config.service
===================================================================
--- cloud-init-0.6.3-bzr532.orig/systemd/cloud-config.service
+++ cloud-init-0.6.3-bzr532/systemd/cloud-config.service
@@ -8,6 +8,7 @@ Wants=network.target
 Type=oneshot
 ExecStart=/usr/bin/cloud-init-cfg all config
 RemainAfterExit=yes
+TimeoutSec=0
 
 [Install]
 WantedBy=multi-user.target
Index: cloud-init-0.6.3-bzr532/systemd/cloud-final.service
===================================================================
--- cloud-init-0.6.3-bzr532.orig/systemd/cloud-final.service
+++ cloud-init-0.6.3-bzr532/systemd/cloud-final.service
@@ -8,6 +8,7 @@ Wants=network.target
 Type=oneshot
 ExecStart=/usr/bin/cloud-init-cfg all final
 RemainAfterExit=yes
+TimeoutSec=0
 
 [Install]
 WantedBy=multi-user.target
Index: cloud-init-0.6.3-bzr532/systemd/cloud-init-local.service
===================================================================
--- cloud-init-0.6.3-bzr532.orig/systemd/cloud-init-local.service
+++ cloud-init-0.6.3-bzr532/systemd/cloud-init-local.service
@@ -7,6 +7,7 @@ After=local-fs.target
 Type=oneshot
 ExecStart=/usr/bin/cloud-init start-local
 RemainAfterExit=yes
+TimeoutSec=0
 
 [Install]
 WantedBy=multi-user.target
Index: cloud-init-0.6.3-bzr532/systemd/cloud-init.service
===================================================================
--- cloud-init-0.6.3-bzr532.orig/systemd/cloud-init.service
+++ cloud-init-0.6.3-bzr532/systemd/cloud-init.service
@@ -8,6 +8,7 @@ Wants=local-fs.target cloud-init-local.s
 Type=oneshot
 ExecStart=/usr/bin/cloud-init start
 RemainAfterExit=yes
+TimeoutSec=0
 
 [Install]
 WantedBy=multi-user.target
