=== modified file 'cloudinit/DataSource.py'
--- cloudinit/DataSource.py	2012-03-19 17:33:39 +0000
+++ cloudinit/DataSource.py	2012-09-14 15:44:22 +0000
@@ -136,7 +136,7 @@
             # make up a hostname (LP: #475354) in format ip-xx.xx.xx.xx
             lhost = self.metadata['local-hostname']
             if is_ipv4(lhost):
-                toks = "ip-%s" % lhost.replace(".", "-")
+                toks = [ "ip-%s" % lhost.replace(".", "-") ]
             else:
                 toks = lhost.split(".")
 

